scripts/check-yaml.py

```diff
@@ -49,8 +51,41 @@ def check(self) -> int:
                         "The \"%s\" file must be non-empty",
                         taxonomy.path.with_name(attribution_path.name),
                     )
+                # NOTE: The following three warnings are intended for the
+                # beta only, at the moment, and only to flag issues for
+                # maintainers to address rather than block on them. We will
+                # revisit when other content is allowed.
+                qna_file_path = taxonomy.rel_path.with_name("qna.yaml")
+                if "knowledge" in qna_file_path.parts:
+                    qna_file_contents = parser.parse(qna_file_path).contents
+                    for element in qna_file_contents["document"]["patterns"]:
+                        if not re.match('.*.md', element):
+                            taxonomy.warning(
+                                "The document \"%s\" should be a markdown "
+                                "file.",
+                                element
+                            )
+                    if not re.match(
+                            'https://github\.com\/.*',
+                            qna_file_contents["document"]["repo"]):
+                        taxonomy.warning(
+                            "The document repo \"%s\" needs to be a "
+                            "GitHub-based repository.",
```

Comment:
No it doesn't have to be GitHub. Any valid git repo could be used. We just expect that any such git repo can be accessed because any necessary authorization is configured.

Reply:
I've been getting conflicting information on this as I've onboarded onto the project, so I'm validating that with oversight first (left a note in the taxonomy triage channel on InstructLab Slack to get more info). I'll update this when I get an answer there.